File: problems/96A/96A.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAXLINE 110

void solve(char *team, int n) {
    int consecutive = 1;
    for (int i = 1; i < n; i++) {
        if (team[i] == team[i-1])
            consecutive++;
        else
            consecutive = 1;
        if (consecutive >= 7) {
            printf("YES\n");
            return;
        }
    }
    printf("NO\n");
}

/* read_string: reads string from stdin into buf with max size n */
char *read_string(char *buf, int n) {
    if (fgets(buf, n, stdin) == NULL) {
        fprintf(stderr, "failed to read line into buffer");
        exit(1);
    }
    buf[strcspn(buf, "\r\n")] = 0;

    return buf;
}

int main(void) {
    char buf[MAXLINE];
    read_string(buf, MAXLINE - 1);
    solve(buf, strlen(buf));
    return 0;
}

File: problems/96A/96A.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
)

func solve(s string) {
	consecutive := 1
	for i := 1; i < len(s); i++ {
		if s[i] == s[i-1] {
			consecutive++
		} else {
			consecutive = 1
		}
		if consecutive == 7 {
			fmt.Println("YES")
			return
		}
	}
	fmt.Println("NO")
}

// ReadString parses next line of stdin to string.
func ReadString() string {
	s := bufio.NewScanner(os.Stdin)
	if ok := s.Scan(); !ok {
		log.Fatal("failed to read from stdin")
	}

	if err := s.Err(); err != nil {
		log.Fatalf("error reading from stdin: %v\n", err)
	}

	return s.Text()
}

func main() {
	s := ReadString()
	solve(s)
}
